# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmIonCompile.cpp
# Commit: aca2e6cff6b9
# Full Hash: aca2e6cff6b9c66232538f82538d3d1965be4eca
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2019-10-22 21:43:14
# Regressor Bug: 1578418
# File Overlap Count: 1
# Description:
#   Bug 1578418 - Use WasmABIResults iterator to place block and function results r=luke,lth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44477
# ==============================================================================

diff -r e21ad27bfd0a -r aca2e6cff6b9 js/src/wasm/WasmIonCompile.cpp
--- a/js/src/wasm/WasmIonCompile.cpp	Tue Oct 22 10:57:43 2019 +0000
+++ b/js/src/wasm/WasmIonCompile.cpp	Tue Oct 22 09:04:59 2019 +0000
@@ -1764,11 +1764,13 @@
 }
 
 static bool EmitBlock(FunctionCompiler& f) {
-  return f.iter().readBlock() && f.startBlock();
+  ResultType params;
+  return f.iter().readBlock(&params) && f.startBlock();
 }
 
 static bool EmitLoop(FunctionCompiler& f) {
-  if (!f.iter().readLoop()) {
+  ResultType params;
+  if (!f.iter().readLoop(&params)) {
     return false;
   }
 
@@ -1784,8 +1786,9 @@
 }
 
 static bool EmitIf(FunctionCompiler& f) {
+  ResultType params;
   MDefinition* condition = nullptr;
-  if (!f.iter().readIf(&condition)) {
+  if (!f.iter().readIf(&params, &condition)) {
     return false;
   }
 
@@ -1799,9 +1802,10 @@
 }
 
 static bool EmitElse(FunctionCompiler& f) {
-  ResultType thenType;
+  ResultType paramType;
+  ResultType resultType;
   DefVector thenValues;
-  if (!f.iter().readElse(&thenType, &thenValues)) {
+  if (!f.iter().readElse(&paramType, &resultType, &thenValues)) {
     return false;
   }
 